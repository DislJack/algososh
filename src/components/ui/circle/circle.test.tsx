import { ElementStates } from "../../../types/element-states";
import { Circle } from "./circle";
import renderer from 'react-test-renderer'

describe('Тестируем компонент <Circle />', () => {
  it('Тестирование рендеринга без букв', () => {
    const circle = renderer
    .create(<Circle />)
    .toJSON();
    expect(circle).toMatchSnapshot();
  });

  it('Тестирование рендеринга с буквами', () => {
    const circle = renderer
    .create(<Circle letter="фыв" />)
    .toJSON();
    expect(circle).toMatchSnapshot();
  });

  it('Тестирование ренеринга с head и tail', () => {
    const circle = renderer
    .create(<Circle letter="авы" head={'head'} tail={'tail'} />)
    .toJSON();
    expect(circle).toMatchSnapshot();
  });

  it('Тестирование рендеринга с react-элементом в head и tail, и пропом isSmall', () => {
    const circle = renderer
    .create(<Circle letter="gfs" head={<Circle letter="hty" isSmall />} tail={<Circle letter="hgyu" isSmall />}  />)
    .toJSON();
    expect(circle).toMatchSnapshot();
  });

  it('Тестирование рендеринга с index', () => {
    const circle = renderer
    .create(<Circle index={2} />)
    .toJSON();
    expect(circle).toMatchSnapshot();
  });

  it('Тестирование рендеринга в состоянии default', () => {
    const circle = renderer
    .create(<Circle state={ElementStates.Default} />)
    .toJSON()
    expect(circle).toMatchSnapshot();
  })

  it('Тестирование рендеринга в состоянии changing', () => {
    const circle = renderer
    .create(<Circle state={ElementStates.Changing} />)
    .toJSON()
    expect(circle).toMatchSnapshot();
  })

  it('Тестирование рендеринга в состоянии modified', () => {
    const circle = renderer
    .create(<Circle state={ElementStates.Modified} />)
    .toJSON();
    expect(circle).toMatchSnapshot();
  })
})