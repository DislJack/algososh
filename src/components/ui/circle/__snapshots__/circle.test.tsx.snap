// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тестируем компонент <Circle /> Тестирование рендеринга без букв 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тестируем компонент <Circle /> Тестирование рендеринга в состоянии changing 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   changing"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тестируем компонент <Circle /> Тестирование рендеринга в состоянии default 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тестируем компонент <Circle /> Тестирование рендеринга в состоянии modified 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   modified"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тестируем компонент <Circle /> Тестирование рендеринга с index 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  >
    2
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
  />
</div>
`;

exports[`Тестируем компонент <Circle /> Тестирование рендеринга с react-элементом в head и tail, и пропом isSmall 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle  small default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        >
          hty
        </p>
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      gfs
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle  small default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        >
          hgyu
        </p>
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
</div>
`;

exports[`Тестируем компонент <Circle /> Тестирование рендеринга с буквами 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      фыв
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тестируем компонент <Circle /> Тестирование ренеринга с head и tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
  >
    head
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      авы
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
  >
    tail
  </div>
</div>
`;
